From 3cc1e3d5ca9b0429f4f1e6618c4fbabac0cfb38b Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 30 Nov 2021 09:32:21 +0100
Subject: [PATCH] BUILD: evports: remove a leftover from the dead_fd cleanup

Commit b1f29bc62 ("MINOR: activity/fd: remove the dead_fd counter") got
rid of FD_UPDT_DEAD, but evports managed to slip through the cracks and
wasn't cleaned up, thus it doesn't build anymore, as reported in github
issue #1467. We just need to remove the related lines since the situation
is already handled by the remaining conditions.

Thanks to Dominik Hassler for reporting the issue and confirming the fix.

This must be backported to 2.5 only.
diff -wpruN '--exclude=*.orig' a~/src/ev_evports.c a/src/ev_evports.c
--- a~/src/ev_evports.c	1970-01-01 00:00:00
+++ a/src/ev_evports.c	1970-01-01 00:00:00
@@ -250,10 +250,6 @@ static void _do_poll(struct poller *p, i
 		 */
 		ret = fd_update_events(fd, n);
 
-		/* If the FD was already dead , skip it */
-		if (ret == FD_UPDT_DEAD)
-			continue;
-
 		/* disable polling on this instance if the FD was migrated */
 		if (ret == FD_UPDT_MIGRATED) {
 			if (!HA_ATOMIC_BTS(&fdtab[fd].update_mask, tid))
